src/v/transform/logging/log_manager.cc

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "transform/logging/log_manager.h"
+
+#include "base/vassert.h"
+#include "config/configuration.h"
+#include "model/namespace.h"
+#include "random/simple_time_jitter.h"
+#include "strings/utf8.h"
+#include "transform/logging/io.h"
+#include "transform/logging/logger.h"
+
+#include <seastar/core/condition-variable.hh>
+#include <seastar/core/manual_clock.hh>
+#include <seastar/core/smp.hh>
+#include <seastar/coroutine/as_future.hh>
+
+namespace transform::logging {
+namespace {
+using namespace std::chrono_literals;
+} // namespace
+
+namespace detail {
+
+template<typename ClockType>
+class flusher {
+    static constexpr ClockType::duration jitter_amt{50ms};
+    static constexpr int MAX_CONCURRENCY = 10;
+
+public:
+    flusher() = delete;
+    explicit flusher(
+      ss::abort_source* as,
+      transform::logging::client* c,
+      config::binding<std::chrono::milliseconds> fi)
+      : _as(as)
+      , _client(c)
+      , _interval_ms(std::move(fi))
+      , _jitter(_interval_ms(), jitter_amt) {
+        _interval_ms.watch([this]() {
+            _jitter = simple_time_jitter<ClockType>{_interval_ms(), jitter_amt};
+        });
+    }
+
+    template<typename BuffersT>
+    ss::future<> start(ss::gate& gate, BuffersT* bufs) {
+        ssx::spawn_with_gate(
+          gate, [this, bufs]() -> ss::future<> { return flush_loop(bufs); });
+        return ss::now();
+    }
+
+    void wakeup() { _wakeup_signal.signal(); }
+
+    ss::future<> stop() {
+        _wakeup_signal.broken();
+        return ss::now();
+    }
+
+    ss::future<> flush_loop(auto* bufs) {
+        while (!_as->abort_requested()) {
+            try {
+                // the duration overload passes now() + dur to the timepoint
+                // overload, but the now() calculation is tied to an underlying
+                // clock type, so it doesn't work with ss::manual_clock. the
+                // timepoint overload template has a clocktype parameter, so we
+                // use that one to get the behavior we want for testing
+                if constexpr (std::is_same_v<ClockType, ss::manual_clock>) {
+                    co_await _wakeup_signal.wait(
+                      ClockType::now() + _jitter.base_duration());
+                } else {
+                    co_await _wakeup_signal.wait(_jitter());
+                }
```

Comment:
Does it simplify things if we always do:

```
co_await _wakeup_signal.wait<ClockType>(ClockType::now() + _jitter.base_duration());
```

Because I think the `else` version you're using here uses `timer::clock::now()` which defaults to steady_clock, which lowres should be fine here.

Reply:
oh, yeah you're absolutely right. I got mixed up associating the `Clock` template param with the class rather than the wait methods.